### ISSUE TITLE
css.properties.visibility - "Collapse" should be listed as properly working in Chrome

### ISSUE THREAD

**Issue by joswhite** [mdn/browser-compat-data#24185]
### What type of issue is this?

Incorrect support data

### What information was incorrect, unhelpful, or incomplete?

"Chrome treats visibility: collapse like hidden, leaving a white gap. Chrome supports the collapse value on `<tr>`, `<thead>`, `<tbody>`, and `<tfoot>` but not on `<col>` and `<colgroup>` elements."

### What browsers does this problem apply to, if applicable?

Chrome 62+.

### What did you expect to see?

Full support for table elements (tr, thead, tbody, tfoot, col, and colgroup), starting in Chrome 92. Chrome isn't leaving a white gap anymore. Partial support for Chrome 62-91.

### Did you test this? If so, how?

I tested it on BrowserStack, using my jsfiddle (https://jsfiddle.net/joseph_white3/xp2kgo5e/). Results:

Before Chrome 62: tr, thead, tbody, and tfoot are treated like "hidden", leaving a white gap. col and colgroup are not supported.
Chrome 62-91: tr, thead, tbody, and tfoot work properly now. col and colgroup are not supported.
Chrome 92+: Full support for table elements (tr, thead, tbody, tfoot, col, and colgroup).

![image](https://github.com/user-attachments/assets/a3ede86d-b52f-4d54-a453-36c2303d2822)

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

https://developer.mozilla.org/en-US/docs/Web/CSS/visibility

### Do you have anything more you want to share?

Here's a screenshot of the current incorrect compatibility table:

![image](https://github.com/user-attachments/assets/419271a4-8c02-4d60-8397-7b5d78de76ad)

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/visibility

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `css.properties.visibility`
* Report started: 2024-08-20T23:53:00.850Z

</details>

**Comment by thx1111** [comment]
I agree - very strongly - that describing chrome/chromium has having "full support" for `visibility: collapse;` is just plain _wrong_.  "Pretending" that "treats visibility: collapse like hidden, leaving a white gap",  as expressed at https://caniuse.com/mdn-css_properties_visibility_collapse , is, in any way, some "version" of `visibility: collapse;` is simply dishonest, as if Google should be given some special dispensation.  The bug report on this issue, "visibility:collapse does not work correctly for table columns or rows", https://issues.chromium.org/issues/40300930 , has been open now for over 10 years, so, apparently Google has no intention of ever fixing their engine.

`visibility: hidden;` is nothing like `visibility: collapse;`.  And this, not even including `visibility: collapse;` support for HTML elements generally, rather than just for certain `table` elements.

Ironically, though, the Microsoft "Edge" browser, apparently also using the Blink browser engine, is listed as having `visibility: collapse;` support without any qualification.  I cannot confirm this one way or the other, as I have never used the Edge browser.

A more honest way to show this `visibility: collapse;` feature on mdn for Blink based browsers would be as having "x No" support, while including the same "See implementation notes."

At least, with Firefox, it is possible to get the same `visibility: collapse;` support for _any_ HTML element generally by wrapping the element inside a gratuitous inline `<table><tr class="collapse"><td><...> ... </...></td></tr></table>`.

By the way, it is quite confusing to determine exactly "how" or "where" to "See implementation notes".  It would be polite to expand that footnote to read, for instance, as "See implementation notes by selecting the box in the table."

**Comment by joswhite** [comment]
Sorry for the confusion, I had some incomplete and missing content in my post. I updated it.

@thx1111 - The bug you mentioned is using "visiblity", which is misspelled. That does make things confusing to test. ;) Chrome actually did fix it.

![image](https://github.com/user-attachments/assets/a91923ba-a2c6-4d2e-a33b-cf1ffd1584d8)

Based on my tests, the latest Chrome has full support for "visibility: collapse". Please let me know if you still disagree. Thanks!

**Comment by thx1111** [comment]
I am looking at the page https://developer.mozilla.org/en-US/docs/Web/CSS/visibility , running the "Try it" demo.  On Firefox, I see that visibility: collapse; removes the "Hide me" box from the layout.  However, on Chromium, Version 135.0.7049.84, on Linux, that same demo page displays  visibility: collapse; the same as  visibility: hidden;, and does not change the layout.  This does not look "fixed" to me - even from the MDN website.

By the way, for what I was doing, I realized that I could use visibility: hidden; instead of visibility: collapse;.  visibility: collapse; is important when dynamic layout is desired.  But visibility: hidden; alone is sufficient when only controlling *where* in the document certain content is displayed, by using the CSS class selector and javascript.

**Comment by caugner** [comment]
Thanks for the report, the explanation for the fix in Chrome 92 seems to be that [TablesNG](https://issues.chromium.org/issues/40625453), a new table layout algorithm, [shipped around that time](https://issues.chromium.org/issues/40625453#comment295).

However, we should mention the issue described by @thx1111.

There are two related Chromium issue reports, but they don't explain why the MDN demo doesn't work:

- [col visibility:collapse does not hide all types of content](https://issues.chromium.org/issues/40802067)
- [Floated / relpos content inside visibility:collapse table column displayed](https://issues.chromium.org/issues/396060182)

**Comment by caugner** [comment]
Looking at [the spec](https://drafts.csswg.org/css-display/#valdef-visibility-collapse): 

> Indicates that the box is collapsed, which can cause it to take up less space than otherwise in a formatting-context–specific way. See [dynamic row and column effects in tables](https://www.w3.org/TR/CSS2/tables.html#dynamic-effects) [[CSS2]](https://drafts.csswg.org/css-display/#biblio-css2) and [collapsed flex items](https://www.w3.org/TR/css-flexbox-1/#visibility-collapse) in flex layout [[CSS-FLEXBOX-1]](https://drafts.csswg.org/css-display/#biblio-css-flexbox-1). In all other cases, however, (i.e. unless otherwise specified) this simply makes the box [invisible](https://drafts.csswg.org/css-display/#invisible), just like [hidden](https://drafts.csswg.org/css-display/#valdef-visibility-hidden).

It would probably be useful to add two subfeatures under `css.properties.visibility.collapse`:

1. "Collapse table elements" with spec_url set to https://www.w3.org/TR/CSS2/tables.html#dynamic-effects
2. "Collapse flex items" with spec_url set to https://www.w3.org/TR/css-flexbox-1/#visibility-collapse

Chrome and Safari partially support collapsing flex items, and there is [an open Safari bug](https://bugs.webkit.org/show_bug.cgi?id=70782), but I couldn't find a corresponding one for Chromium:

- Chrome and Safari both properly render [the "Example 4" in the spec](https://www.w3.org/TR/css-flexbox-1/#visibility-collapse).
- But they don't render [the example on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#try_it).

@thx1111 Could you please file [a Chromium issue](https://issues.chromium.org/issues) about collapsing flex items, pointing to the MDN demo? 🙏

**Comment by thx1111** [comment]
Added Issue: https://issues.chromium.org/issues/413595082

In a tragically amusing irony, a different "visibility: collapse;" issue was filed six hours ago, "`visibility` css transition effects different than before", https://issues.chromium.org/issues/413427438 , and then, nonetheless, arguing - after the fact - that "This seems like a corner case that doesn't justify it being release blocking.", Chromium version 136 was released yesterday.

**Comment by davidsgrogan** [comment]
> Added Issue: https://issues.chromium.org/issues/413595082

> In a tragically amusing irony, a different "visibility: collapse;" issue was filed six hours ago, "`visibility` css transition effects different than before", https://issues.chromium.org/issues/413427438

No, at its core that new issue is NOT specific to `visibility:collapse`. Look at the more recent comments in that bug. The bug reproduces with `visibility:hidden`. `collapse` is irrelevant.

Chrome support for `visibility:collapse` is definitely not "x No". Chrome supports it for tables and all table parts, which was the most common requested use case and and has supported it for years, as the OP here correctly claims.

The claim above that
> The bug report on this issue, "visibility:collapse does not work correctly for table columns or rows", https://issues.chromium.org/issues/40300930 , *has been open now for over 10 years*, so, apparently Google has no intention of ever fixing their engine.

...  is patently incorrect. The bug was closed in 2021.

Chrome does NOT support the `flex`-specific and ruby-specific `visibility:collapse` behavior, for which there has been low author demand.

But for ALL other elements, chrome correctly treats `visibility:collapse` as `visibility:hidden`, which https://issues.chromium.org/issues/413595082 seems to not understand.

**Comment by joswhite** [comment]
Thanks everyone! #26613 has been merged so I'll go ahead and close this now

**Comment by davidsgrogan** [comment]
Great, thanks for updating.
